File: MultiLanguageSandbox/src/test/deps/php/43.php
<?php

/**
 * Finds the length of the longest sequence of consecutive numbers in an array.
 * Consecutive numbers are numbers which come one after the other in sequence without any gaps.
 * For example, in the array [1, 3, 5, 4, 2, 6], the longest sequence of consecutive numbers is [1, 2, 3, 4, 5, 6], which has a length of 6.
 * Input: An array of integers.
 * Output: The length of the longest sequence of consecutive numbers found in the array.
 */
function findLongestConsecutiveSequence(array $numbers): int {

    if (empty($numbers)) {
        return 0;
    }
    
    $numberSet = array_flip($numbers); // Using array_flip for O(1) lookups
    $maxLength = 0;
    
    foreach ($numberSet as $num => $value) {
        // Only start counting if the number is the beginning of a sequence
        if (!isset($numberSet[$num - 1])) {
            $currentNum = $num;
            $currentLength = 1;
            
            // Count how many consecutive numbers follow
            while (isset($numberSet[$currentNum + 1])) {
                $currentNum++;
                $currentLength++;
            }
            
            $maxLength = max($maxLength, $currentLength);
        }
    }
    
    return $maxLength;
}
function main(){
    assert(findLongestConsecutiveSequence([1, 5, 6, 2, 3, 4, 5, 6, 8, 9]) === 6);
    assert(findLongestConsecutiveSequence([10, 4, 20, 1, 3, 2]) === 4);
    assert(findLongestConsecutiveSequence([1, 2, 2, 3]) === 3);
    assert(findLongestConsecutiveSequence([]) === 0);
    assert(findLongestConsecutiveSequence([7]) === 1);
    
}


main();
?>

File: MultiLanguageSandbox/src/test/deps/C#/43.cs
using System;
using System.Collections.Generic;
using System;
using System.Collections.Generic;
using System.Diagnostics;

class Program
{

/* Finds the length of the longest consecutive elements sequence in a given list of integers.
    The sequence does not need to be in order but must consist of consecutive natural numbers.
    >>> LongestConsecutiveSequence(new List<int> {1, 5, 6, 2, 3, 4, 5, 6, 8, 9})
    6
    >>> LongestConsecutiveSequence(new List<int> {10, 4, 20, 1, 3, 2})
    4
*/

static int LongestConsecutiveSequence(List<int> sequence)
{
        if (sequence == null || sequence.Count == 0)
        {
            return 0;
        }

        HashSet<int> numSet = new HashSet<int>(sequence);
        int maxLength = 0;

        foreach (int num in numSet)
        {
            // Check if it's the start of a sequence
            if (!numSet.Contains(num - 1))
            {
                int currentNum = num;
                int currentLength = 1;

                while (numSet.Contains(currentNum + 1))
                {
                    currentNum++;
                    currentLength++;
                }

                maxLength = Math.Max(maxLength, currentLength);
            }
        }

        return maxLength;
    }
    static void Main()
    {
        Debug.Assert(LongestConsecutiveSequence(new List<int> {1, 5, 6, 2, 3, 4, 5, 6, 8, 9}) == 6);
        Debug.Assert(LongestConsecutiveSequence(new List<int> {10, 4, 20, 1, 3, 2}) == 4);
        Debug.Assert(LongestConsecutiveSequence(new List<int> {100, 4, 200, 1, 3, 2}) == 4);
        Debug.Assert(LongestConsecutiveSequence(new List<int> {1, 1}) == 1);

    }
}

